pkg/cli/cmd/directory/check.go

```diff
@@ -0,0 +1,77 @@
+package directory
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/aserto-dev/clui"
+	"github.com/aserto-dev/go-directory/aserto/directory/reader/v3"
+	"github.com/aserto-dev/topaz/pkg/cli/cc"
+	"github.com/aserto-dev/topaz/pkg/cli/clients"
+	"github.com/aserto-dev/topaz/pkg/cli/jsonx"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type CheckCmd struct {
+	Request  string `arg:""  type:"string" name:"request" optional:"" help:"json request or file path to check permission request or '-' to read from stdin"`
+	Template bool   `name:"template" help:"prints a check permission request template on stdout"`
+	clients.Config
+}
+
+func (cmd *CheckCmd) Run(c *cc.CommonCtx) error {
+	if cmd.Template {
+		return printCheckPermissionRequest(c.UI)
+	}
+
+	client, err := clients.NewDirectoryClient(c, &cmd.Config)
+	if err != nil {
+		return errors.Wrap(err, "failed to get directory client")
+	}
+
+	if cmd.Request == "" {
+		return errors.New("request argument is required")
+	}
+
+	var req reader.CheckRequest
+	if cmd.Request == "-" {
```

Comment:
This logic is duplicated in all the directory commands. I think it can be pulled out into a single generic function:
```go
type Message[T any] interface {
	proto.Message
	*T
}

func UnmarshalRequest[T any, M Message[T]](src string, msg *M) error {
    ...
}
```